### ISSUE TITLE
Large numbers are cut off at 2147483.647

### ISSUE THREAD

**Issue by YanWittmann** [Skyball2000/j2chartjs#11]

**Comment by YanWittmann** [comment]
```java
LineChartDataset dataset = new LineChartDataset().setBorderColor(new Color(140, 140, 140, 107)).setLabel("Dataset awesome");
LineChartData data = new LineChartData().addDataset(dataset);
for (int i = 0; i < 100; i++) {
    dataset.addData(new Random().nextInt(230512202));
    dataset.addPointBackgroundColor(new Random().nextInt(2) == 1 ? Color.RED : Color.BLUE);
    dataset.addPointRadius(1 + new Random().nextInt(3));
}
ChartOptions options = new ChartOptions().setResponsive(false);
System.out.println(new LineChart().setChartOptions(options).setChartData(data).build());
```

**Comment by YanWittmann** [comment]
This is due to the double limitation. the value has to be checked first:
```java
    private static Object smartAddToJsonForConfigurationConverter(Object object) {
    if (object == null) return null;
    if (object instanceof Color) {
        return convertColorToJs((Color) object);
    } else if (object instanceof Double) {
        return roundToDecimals((Double) object, 3);
    } else if (object instanceof Number) {
        double doubleValue = ((Number) object).doubleValue(); // here
        if (doubleValue >= 2147483) return ((Number) object).longValue();
        return roundToDecimals(((Number) object).doubleValue(), 3);
    } else if (object instanceof JSONObject || object instanceof JSONArray) {
        return object;
    } else if (object instanceof AbstractChartOption) {
        return ((AbstractChartOption) object).toJson();
    } else if (object instanceof ChartData) {
        return ((ChartData) object).toJson();
    } else if (object instanceof MixedChartData) {
        return ((MixedChartData) object).toJson();
    } else if (object instanceof MixedChart) {
        return ((MixedChart) object).toJson();
    } else if (object instanceof AnimationEasingType) {
        return ((AnimationEasingType) object).getKey();
    }
    return object;
}
```

**Comment by YanWittmann** [comment]
Fixed in 68b7ea0853a99ee44b5b8f84eeccbb69a428ef0a
